src/unfold.rs

```diff
@@ -69,16 +68,16 @@ pub fn unfold<T, F, Fut, It>(init: T, f: F) -> Unfold<T, F, Fut>
     }
 }
 
-/// Stream for the [`unfold`] function.
-#[must_use = "streams do nothing unless polled"]
-pub struct Unfold<T, F, Fut> {
-    f: F,
-    state: Option<T>,
-    fut: Option<Fut>,
+pin_project_lite::pin_project!{
+    /// Stream for the [`unfold`] function.
+    #[must_use = "streams do nothing unless polled"]
+    pub struct Unfold<T, F, Fut> {
+        f: F,
+        state: Option<T>,
+        fut: Option<Pin<Box<Fut>>>,
```

Comment:
Holding anonymous futures, like the ones generated by `async fn` and async closures don't implement `Unpin` which means we can't store them in structs. Instead we need to wrap them inside of a `Pin<Box<Fut>>` to store them on the heap. Once we have full async closure support in the language, certain async closures may imply `Unpin`. But we're not there yet unfortunately.